extensions/smallrye-opentracing/runtime/src/main/java/io/quarkus/smallrye/opentracing/runtime/TracingConcurrencyStrategy.java

```diff
@@ -0,0 +1,98 @@
+package io.quarkus.smallrye.opentracing.runtime;
+
+import java.util.concurrent.Callable;
+
+import org.jboss.logging.Logger;
+import org.jboss.logging.Logger.Level;
+
+import com.netflix.hystrix.strategy.HystrixPlugins;
+import com.netflix.hystrix.strategy.concurrency.HystrixConcurrencyStrategy;
+import com.netflix.hystrix.strategy.eventnotifier.HystrixEventNotifier;
+import com.netflix.hystrix.strategy.executionhook.HystrixCommandExecutionHook;
+import com.netflix.hystrix.strategy.metrics.HystrixMetricsPublisher;
+import com.netflix.hystrix.strategy.properties.HystrixPropertiesStrategy;
+
+import io.opentracing.Scope;
+import io.opentracing.ScopeManager;
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+
+/**
+ * This strategy configures Hystrix to propagate tracing context (Spans) across threads.
+ */
+public class TracingConcurrencyStrategy extends HystrixConcurrencyStrategy {
+    private static Logger log = Logger.getLogger(TracingConcurrencyStrategy.class.getName());
+
+    private HystrixConcurrencyStrategy delegateStrategy;
+    private Tracer tracer;
+
+    public static TracingConcurrencyStrategy register(Tracer tracer) {
+        return new TracingConcurrencyStrategy(tracer);
+    }
+
+    private TracingConcurrencyStrategy(Tracer tracer) {
+        this.tracer = tracer;
+        try {
+            this.delegateStrategy = HystrixPlugins.getInstance().getConcurrencyStrategy();
+            if (this.delegateStrategy instanceof TracingConcurrencyStrategy) {
+                return;
+            }
+
+            HystrixCommandExecutionHook commandExecutionHook = HystrixPlugins.getInstance().getCommandExecutionHook();
+            HystrixEventNotifier eventNotifier = HystrixPlugins.getInstance().getEventNotifier();
+            HystrixMetricsPublisher metricsPublisher = HystrixPlugins.getInstance().getMetricsPublisher();
+            HystrixPropertiesStrategy propertiesStrategy = HystrixPlugins.getInstance().getPropertiesStrategy();
+
+            HystrixPlugins.reset();
+            HystrixPlugins.getInstance().registerConcurrencyStrategy(this);
+            HystrixPlugins.getInstance().registerCommandExecutionHook(commandExecutionHook);
+            HystrixPlugins.getInstance().registerEventNotifier(eventNotifier);
+            HystrixPlugins.getInstance().registerMetricsPublisher(metricsPublisher);
+            HystrixPlugins.getInstance().registerPropertiesStrategy(propertiesStrategy);
+        } catch (Exception ex) {
+            log.log(Level.TRACE, "Failed to register " + TracingConcurrencyStrategy.class +
+                    ", to HystrixPlugins", ex);
+        }
+    }
+
+    @Override
+    public <T> Callable<T> wrapCallable(Callable<T> callable) {
+        if (callable instanceof OpenTracingHystrixCallable) {
```

Comment:
Curious why this and the similar check on line 66 are necessary - have you seen scenarios where `OTHystrixCallable` instances are provided?

Reply:
This avoids double wrapping. I haven't seen those scenarios but similar implementations are also checking for this so I guess it's important.